contracts/lib.rs

```diff
@@ -123,28 +214,31 @@ mod open_payroll {
             self.paused_block_at.is_some()
         }
 
-        // ensure_in_not_paused ensures that the contract is not paused
-        fn ensure_in_not_paused(&self) -> Result<(), Error> {
+        // ensure_is_not_paused ensures that the contract is not paused
+        fn ensure_is_not_paused(&self) -> Result<(), Error> {
             if self.is_paused() {
                 return Err(Error::ContractIsPaused);
             }
             Ok(())
         }
 
+        
         /// Add a new beneficiary or modify the multiplier of an existing one.
+        /// TODO: maybe split this function in two
         #[ink(message)]
         pub fn add_or_update_beneficiary(
             &mut self,
             account_id: AccountId,
-            multipliers: Vec<Multiplier>,
+            multipliers: Vec<(MultiplierId,Multiplier)>,
         ) -> Result<(), Error> {
             self.ensure_owner()?;
 
             // Check that the multipliers are valid and have the same length as the base_multipliers
```

Comment:
Add check to ensure that multiplier_id exists in base multipliers and also ensure chat multiplier_ids are not repeated in the vec

Reply:
this is already solved ?